src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2bail/Cas2BailApplicationNoteService.kt

```diff
@@ -0,0 +1,171 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas2bail
+
+import io.sentry.Sentry
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.NewCas2ApplicationNote
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.config.NotifyConfig
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.*
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.cas2bail.*
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.AuthorisableActionResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.CasResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.ValidatableActionResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.EmailNotificationService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.ExternalUserService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.HttpAuthService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.NomisUserService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas2.UserAccessService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.toCas2UiFormat
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.toCas2UiFormattedHourOfDay
+import java.time.OffsetDateTime
+import java.util.*
+
+@Service("Cas2BailApplicationNoteService")
+class Cas2BailApplicationNoteService(
+  private val cas2BailApplicationRepository: Cas2BailApplicationRepository,
+  private val cas2BailAssessmentRepository: Cas2BailAssessmentRepository,
+  private val cas2BailApplicationNoteRepository: Cas2BailApplicationNoteRepository,
+  private val userService: NomisUserService,
+  private val externalUserService: ExternalUserService,
+  private val httpAuthService: HttpAuthService,
+  private val emailNotificationService: EmailNotificationService,
+  private val userAccessService: UserAccessService,
+  private val notifyConfig: NotifyConfig,
+  @Value("\${url-templates.frontend.cas2.application-overview}") private val applicationUrlTemplate: String,
+  @Value("\${url-templates.frontend.cas2.submitted-application-overview}") private val assessmentUrlTemplate: String,
```

Comment:
presumably you'll need cas2bail specific templates? These will need defining in the helm .env files for each environment. This applies to multiple usages in this PR